documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+        description
+            "Initial revision.";
+    }
+
+    container sonic-dash {
+        container DASH_VNET {
+            description "DASH VNET configuration, part of APP_DB";            
+
+            list DASH_VNET_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern "Vnet[a-zA-Z0-9_-]+" {
+                            error-message "Invalid Vnet name";
+                            error-app-tag vnet-name-invalid;
+                        }
+                    }
+                }   
+
+                leaf vni {
+                    type  uint32 {
+                        range 0..16777215;
+                    }
+                }
+
+                leaf guid {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf vxlan_tunnel {
+                    description "Vxlan tunnel name"
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf-list address_spaces {
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+
+            } /* end of list DASH_VNET_LIST */
+        } /* end of container DASH_VNET */
+
+        container DASH_QOS {
+            description "DASH QOS configs to specify bandwidth, cps, part of APP_DB";            
+
+            list DASH_QOS_LIST {
+                key "name";
+
+                leaf name{
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf bw {
+                    description "Bandwidth in kbps"
+                    type uint64 {
+                        range "0..100000000" {
+                            error-message "kbps value out of range";
+                        }
+                    }
+                }
+
+                leaf cps {
+                    description "Max number of connections per second"
+                    type uint64 {
+                        range "0..100000000" {
+                            error-message "cps value out of range";
+                        }
+                    }
+                }
+
+                leaf flows {
+                    description "Number of flows"
+                    type uint64 {
+                        range "0..100000000" {
+                            error-message "flows value out of range";
+                        }
+                    }
+                }
+
+            } /* end of list DASH_QOS_LIST */
+        } /* end of container DASH_QOS */
+        
+        container DASH_ENI {
+            description "DASH ENI object to specify eni params; part of APP_DB";            
+
+            list DASH_ENI_LIST {
+                key "name";
```

Comment:
Do we need a reserved VNI table for determining the traffic direction (VM traffic or Network traffic)? and then we can lookup the following table to get the corresponding ENI.

Outbound (VM -> network):
   [Reserved VNI, src-MAC] -> ENI

Inbound (Network -> VM):
   [Non-Reserved VNI, dst-MAC] -> ENI